department-of-veterans-affairs/caseflow
# Add Frontend changes for marking cases overtime

Connects #13415 

### Description
Frontend work for #13415

Features added:  
- modal to mark/remove overtime status from case 
- button on Case Details page 

### Acceptance Criteria
- [ ] Code compiles correctly and tests pass
- [ ] Work is behind the feature toggle: :overtime_revamp
- [ ] The feature should be accessible to the following user groups: VLJs (SSCs are VLJs in Caseflow)
- [ ] When judges or SSCs assign cases to attorneys, they can mark the case as overtime for both AMA and legacy cases
- [ ] Judges or SSCs have the ability to edit the OT designation after assignment

### Testing Plan

1. enable `overtime_revamp` feature toggle 
```ruby
FeatureToggle.enable!(:overtime_revamp)
```
2. Go to http://localhost:3000/queue as BVAAABSHIRE
3. Select a legacy appeal
4. Ensure you can see the option to Mark this case as overtime
5. Select the option and confirm in the modal the case is to be marked as overtime
6. When the page reloads, ensure that you see the Success banner at the top of the page and you can now see the option to Remove this case from overtime
7. Select the updated option and confirm in the modal the overtime status is to be removed from the case 
8. When the page reloads, ensure that you see the Success banner at the top of the page and you can now see the original option to Mark this case as overtime
9. Repeat steps for an ama appeal
10. Assign one legacy and one ama case each to an attorney on BVAAABSHIRE's team
11.  Ensure that you can successfully repeat steps 4 - 9
12.  Login as whichever attorney you assigned the legacy and ama cases to
13.  Ensure you do not have the option to update the overtime status of the respective cases

### User Facing Changes
#### Legacy assigned to judge 
#### Before

<img width="1438" alt="Screen Shot 2020-05-14 at 10 19 32 AM" src="https://user-images.githubusercontent.com/18618189/81945714-72cb1400-95cc-11ea-8d1a-ca350cf11251.png">

#### After

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/18618189/81947782-f259e280-95ce-11ea-824b-a679f3a52019.gif)

#### AMA assigned to judge 
#### Before

<img width="1438" alt="Screen Shot 2020-05-14 at 10 25 32 AM" src="https://user-images.githubusercontent.com/18618189/81946419-54194d00-95cd-11ea-8b55-9e4554780b73.png">

#### After

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/18618189/81946999-081ad800-95ce-11ea-8c7e-ffd2c1896f16.gif)

#### Legacy assigned to attorney
#### Before

<img width="1440" alt="Screen Shot 2020-05-14 at 2 13 17 PM" src="https://user-images.githubusercontent.com/18618189/81970778-0b25c080-95ee-11ea-997c-a410574bd0bb.png">

#### Legacy assigned to attorney
#### After

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/18618189/82260130-b30ef700-992a-11ea-8509-e4b60bef7db2.gif)


#### AMA assigned to attorney
#### Before

<img width="1440" alt="Screen Shot 2020-05-14 at 2 13 36 PM" src="https://user-images.githubusercontent.com/18618189/81970810-17118280-95ee-11ea-8181-4f8b9ed1a3c3.png">

#### AMA assigned to attorney
#### After

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/18618189/82260377-26b10400-992b-11ea-8269-7f9041ab2e96.gif)
